homeassistant/components/vera/config_flow.py

```diff
@@ -0,0 +1,110 @@
+
+import pyvera as pv
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EXCLUDE, CONF_LIGHTS
+from homeassistant.core import callback
+
+from .const import CONF_CONTROLLER, DOMAIN
+
+LIST_REGEX = re.compile("[^0-9]+")
+
+
+def str_to_int_list(data: str) -> List[int]:
+    """Convert a string to an int list."""
+    return [int(s) for s in LIST_REGEX.split(data) if len(s) > 0]
+
+
+def int_list_to_str(data: List[int]) -> str:
+    """Convert an int list to a string."""
+    return " ".join([str(i) for i in data])
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Options for the component."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Init object."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title="",
+                data={
+                    CONF_LIGHTS: str_to_int_list(user_input.get(CONF_LIGHTS, "")),
+                    CONF_EXCLUDE: str_to_int_list(user_input.get(CONF_EXCLUDE, "")),
+                },
+            )
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_LIGHTS,
+                        default=int_list_to_str(
+                            self.config_entry.options.get(CONF_LIGHTS, [])
+                        ),
+                    ): str,
+                    vol.Optional(
+                        CONF_EXCLUDE,
+                        default=int_list_to_str(
+                            self.config_entry.options.get(CONF_EXCLUDE, [])
+                        ),
+                    ): str,
+                }
+            ),
+        )
+
+
+class VeraFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Vera config flow."""
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry) -> OptionsFlowHandler:
+        """Get the options flow."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input: dict = None):
+        """Handle user initiated flow."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason="already_setup")
+
+        if user_input is not None:
+            return await self.async_step_finish(user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Required(CONF_CONTROLLER): str}),
+        )
+
+    async def async_step_import(self, config: dict):
+        """Handle a flow initialized by import."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason="already_setup")
```

Comment:
If we update the existing entry options here instead, we have both the existing entry data and new config yaml data available to compare and can decide if we should update and overwrite entry options or not.

Reply:
I think the more consistent option would be to disable options editing when configured from configuration.yml. If they are configuring from the file, then we should always honor that and not overwrite with UI provided values at all. It looks like HA's direction for configuration is 100% UI based, so discouraging/deprecating this feature would fit. Thoughts?

Reply:
If we expose options in the GUI, changes made there should not be overwritten just because the user restarts home assistant. The latest set options should apply.

You can decide if you want to expose options in config yaml, in the GUI or both. But the rule above still applies.

Having options both in yaml and GUI requires more code maintenance, so it's fair if you decide to limit it to one way.